tests/integration/kubernetes/k8s-confidential-attestation.bats

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bats
+# Copyright 2024 IBM Corporation
+# Copyright 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+export test_key="aatest"
+export SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
```

Comment:
I was looking at the logic below to enable/disable the attestation tests, and I was considering to rely on a single variable ($KBS) instead of the list of hypervisors. I explain...

In https://github.com/kata-containers/kata-containers/pull/9116 it changes the k8s for AKS job to get KBS installed. The jobs that got KBS installed will have the `KBS=true` environment variable exported. This avoid us from having in two places (here and in the workflow yamls) the list of hypervisors that should have attesattion tests ran.

Thus, the checker would be just:
```shell
if [ "${KBS:-false}" = "false" ]; then
 skip "Test not supported for ${KATA_HYPERVISOR}"
fi
```

What do you think?

Reply:
@wainersm so the KBS=true where is going to be defined? because for example https://github.com/kata-containers/kata-containers/blob/main/.github/workflows/run-k8s-tests-on-aks.yaml#L48 here we have the env variables but that will be applied for all hypervisors

Reply:
@GabyCT yes, the workflow will have `KBS=true` exported. However, ignore my suggestion...let's keep it as is, i.e., using the `SUPPORTED_TEE_HYPERVISORS` and `SUPPORTED_NONTEE_HYPERVISORS` which seems the "standard" today. In a future I should be proposing something different to handle the selection of confidential tests on NON-TEE jobs; for now let's keep it as is.

Reply:
Well it still might be a good idea to add the `KBS` check on top of the lists of supported ones to ensure we don't execute this test when KBS is not configured, what do you think? (@wainersm mentioned we might include KBS only for some of the instance type(s). It should be filtered using the K8S_TEST_SMALL_HOST_UNION/K8S_TEST_NORMAL_HOST_UNION lists but still it might be a good idea to check for that variable)